### PR TITLE
minor: Use case docs

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

File: CONCEPTS.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Indexing
+
+Indexing your documents is a vital part of any production RAG application. In short, indexing allows for your documents to be stored, and searchable to prevent duplicate documents from being stored. This is important for a few reasons:
+
+1. **Duplicate Results**: Say you update your vector store without using an Indexing API. Now you may have two identical documents in your store. Then, when you perform a semantic search, instead of getting K number of different results, you'll get duplicates as the semantic search only returns documents which are semantically close to the query.
+
+2. **Performance**: Indexing your documents allows for faster ingestion. With indexing you don't have to generate embeddings for every document on ingestion, and instead only need to generate embeddings for new documents.
+
+The indexing API also uses a Record Manager to store the records of previously indexed documents in between ingestion. This manager stores the hashed values of the documents, and the time they were ingested. This allows for the indexing API to only ingest new documents, and not duplicate documents.
```
Comment (hwchase17) [inline CONCEPTS.md:27]: you mention this, without introducting this. eg Indexing as a concept != Indexing API neccessarily. I'd maybe add a sentence introducing it - "in order to help with indexing we use LangChain Indexing API, etc etc

File: CONCEPTS.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Query Analysis
```
Comment (hwchase17) [inline CONCEPTS.md:29]: id maybe comment on why we only do this for follow up quetions and not the first message

File: DEPLOYMENT.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
+# Deployment
+
+For optional DX when deploying Chat LangChain JS, you should use Vercel for the frontend & edge API, and GitHub action for the recurring ingestion tasks.
```
Comment (hwchase17) [inline DEPLOYMENT.md:3]: optional or optimal? Also unclear if this is optimal DX, just the one we set up

File: DEPLOYMENT.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+These should be the same secrets as were added to Vercel.
+
+Next, navigate to the "Actions" tab and confirm you understand your workflows, and enable them.
+
+Then, click on the "Update index" workflow, and click "Enable workflow". Finally, click on the "Run workflow" dropdown and click "Run workflow".
+
+Once this has finished you can visit your production URL from Vercel, and start using the app!
+
+# Running locally
```
Comment (hwchase17) [inline DEPLOYMENT.md:91]: hm i dont love that is this here. id maybe make this its own page? eg if i wanted to figure out how to run something locally, idk if id think to look under deployment

File: LANGSMITH.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+## Observability
+
+Observability is simple when using LangChain as your LLM framework. In its simplest form, all you need is to set two environment variables:
+
+```shell
+LANGCHAIN_TRACING_V2=true
+LANGCHAIN_API_KEY=...
+```
+
+For improved DX, you also sometimes need to pass a config object through to different functions, like so:
```
Comment (hwchase17) [inline LANGSMITH.md:17]: what does this improve? (unclear to reader)

File: LANGSMITH.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+```shell
+yarn add langsmith
+```
+
+Then, you'll want to define some custom criteria to evaluate your dataset on. Some examples are:
+
+- **Semantic similarity**: How similar your generated response is to the ground truth (dataset answers).
+- **LLM as a judge**: Use an LLM to judge and assign a score to your generated response.
+
+Finally, configure your evaluation criteria and use the [`runOnDataset`](https://api.js.langchain.com/functions/langchain_smith.runOnDataset.html) function to evaluate your dataset.
```
Comment (hwchase17) [inline LANGSMITH.md:52]: we should add code for this

File: MODIFY.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+- LangChain.js Documentation
+
+If all you would like to update is update which website(s) to scrape and ingest, you only need to modify/remove these functions:
+
+- `loadLangSmithDocs`
+- `loadAPIDocs`
+- `loadLangChainDocs`
+
+If you want to ingest another way, consult the [document loader](https://js.langchain.com/docs/modules/data_connection/document_loaders/) section of the LangChain docs.
+Using any LangChain.js document loader, you'll be able to easily and efficiently fetch & ingest from a large variety of sources. Additionally, the LangChain.js document loader API will always return documents in the same format ([`DocumentInterface`](https://api.js.langchain.com/interfaces/langchain_core_documents.DocumentInterface.html)) so you do not need to modify the format before adding to your indexing API or vector store.
```
Comment (hwchase17) [inline MODIFY.md:31]: how to modify the vectorstore (eg change provider) should be a focus

**Review by timeowilliams — APPROVED** [review]

File: DEPLOYMENT.md (hunk 4 above)
Comment (bracesproul) [reply DEPLOYMENT.md]: added `RUN_LOCALLY.md`

**Pull request merged**